source/common/upstream/logical_dns_cluster.cc

```diff
@@ -37,6 +38,11 @@ LogicalDnsCluster::LogicalDnsCluster(
       cluster, dns_refresh_rate_ms_.count(), factory_context.random());
 
   const auto& locality_lb_endpoints = load_assignment_.endpoints();
+  if (!locality_lb_endpoints.empty() && locality_lb_endpoints[0].priority() != 0) {
```

Comment:
I don't think zone aware routing make sense at all for logical DNS. Should we just block it entirely? I guess it makes sense for static? Basically I'm just wondering if we should block it for cases in which it would not really work well? WDYT?

Reply:
I think the main concern would be that there are existing users that use zone aware LB that happens to have a logical DNS cluster defined.  

How about we instead rewrite the priority of the LLB that is used to always be 0? Since we enforce that there is ever only one host, the priority value shouldn't matter, so it seems safe to just coerce it into 0 during cluster creation?

Reply:
Yeah that seems reasonable to me and maybe also emit a warning?